doc: add minutes for meeting June 7 2023 (#1398)
nodejs/TSC: meetings/2023-06-07.md

* Robert Nagy @ronag (voting member)
* Ruy Adorno @ruyadorno (voting member)

## Agenda

### Announcements

* Rafael - Node.js 20.3 should be released tomorrow

### CPC and Board Meeting Updates

*Extracted from **tsc-agenda** labeled issues and pull requests from the **nodejs org** prior to the meeting.

* No updates.

### nodejs/node-gyp

* DISCUSSION: Release handling [#2860](https://github.com/nodejs/node-gyp/issues/2860)
  * Michael gave a quick overview of the issue
  * Rafael, one idea might be to ask Foundation to fund rewriting in Node.js
  * Need to ask more broadly if we have any collaborators that have skills/would be interested in getting involved.

### nodejs/node

* module: make CJS load from ESM loader [#47999](https://github.com/nodejs/node/pull/47999)
  * Added to agenda for visibility, not much to discuss
  * Related Geoffrey, Qard added for more guidance on handling breaking changes on experimental features - <https://github.com/nodejs/TSC/issues/1397>

### nodejs/TSC

* CLI tools behind a Nodejs namespace [#1178](https://github.com/nodejs/TSC/issues/1178)
  * leave on agenda until we have Daniele to confirm we have covered

>   * ok to start migrating packages to the node-core namespace.

Reviewer: ```suggestion
  * ok to start migrating packages to the node-core namespace.
  * we need to give TSC members access to npm for deprecating old packages
```

  * Rafael will add comment on 1178 that covers how it was handled with fastify.
    * <https://github.com/fastify/deprecate-modules>
    * Workflow assuming package is currently on version v2.x:
      * Release a new major with npm deprecate pointing to the new module and no content, so version v3.0.0
      * Create a new package with the correct name: @node-core/module in the version v4.0.0

### nodejs/admin

* Archive nodejs/help in favour of a Help Section on Discussions [#802](https://github.com/nodejs/admin/issues/802)
  * Lets defer until a future meeting where ideally we have Gireesh

## Strategic Initiatives

* Yagiz - Recent changes from the performance initiative:

>   * I opened a back-port for Ada v2.5 to LTS (which comes with 100% performance boost)

Reviewer 2: FWIW, I know that this is just quoting another comment verbatim, but we should be careful to make accurate statements about performance improvements. Are _all_ `URL` operations really twice as fast now on all platforms, especially as visible from JS? Maybe, I have no idea, but it has happened before that performance initiatives misrepresented or generalized results of microbenchmarks.

Author: @anonrig can you take a look and suggest updated text for this one and the later comment as well?

Reviewer 3: I'm currently on paternity leave but we can remove all numeric values, and use terms like "significant improvements", "major changes" etc. if not, we can just reduce the changes to "improved the performance of X"

Reviewer 2: ```suggestion
  * I opened a back-port for Ada v2.5 to LTS (which comes with performance improvements)
```

  * Khafra started writing mimesniff which will be included in the next release of Ada. <https://github.com/ada-url/mimesniff>
  * Vinicious opened a pull request for improving the performance of legacy main resolve in module resolution: src,lib: reducing C++ calls of esm legacy main resolve node#48325
  * I opened a pull request for adding NDEBUG=1 to clang flags. Feedback is appreciated. build: add -DNDEBUG=1 to clang flags node#48271

>   * Libuv 1.45.0 - 800% faster filesystem ops got merged into Node and will be included in the next release

Reviewer 3: I mostly shared those numbers for TSC members. We can remove all numeric values concerning benchmarks.

>   * Libuv 1.45.0 - 800% faster filesystem ops got merged into Node and will be included in the next release

Reviewer 2: ```suggestion
  * Libuv 1.45.0 - significantly faster filesystem ops got merged into Node.js and will be included in the next release
```

* CLI tools behind a Nodejs namespace [#1178](https://github.com/nodejs/TSC/issues/1178)
  * leave on agenda until we have Daniele to confirm we have covered
  * ok to start migrating packages to the node-core namespace.
  * we need to give TSC members access to npm for deprecating old packages
  * Rafael will add comment on 1178 that covers how it was handled with fastify.
    * <https://github.com/fastify/deprecate-modules>
    * Workflow assuming package is currently on version v2.x:
      * Release a new major with npm deprecate pointing to the new module and no content, so version v3.0.0
      * Create a new package with the correct name: @node-core/module in the version v4.0.0

### nodejs/admin

* Archive nodejs/help in favour of a Help Section on Discussions [#802](https://github.com/nodejs/admin/issues/802)
  * Lets defer until a future meeting where ideally we have Gireesh

## Strategic Initiatives

* Yagiz - Recent changes from the performance initiative:
  * I opened a back-port for Ada v2.5 to LTS (which comes with 100% performance boost)
  * Khafra started writing mimesniff which will be included in the next release of Ada. <https://github.com/ada-url/mimesniff>

> * Vinícius opened a pull request for improving the performance of legacy main resolve in module resolution: src,lib: reducing C++ calls of esm legacy main resolve node#48325

Reviewer 3: The indentation of this line is wrong

Reviewer 4: ```suggestion
  * Vinícius opened a pull request for improving the performance of legacy main resolve in module resolution: src,lib: reducing C++ calls of esm legacy main resolve node#48325
```

>   * Vinicious opened a pull request for improving the performance of legacy main resolve in module resolution: src,lib: reducing C++ calls of esm legacy main resolve node#48325

Reviewer 5: Vinícius, just a typo :)

>   * Vinicious opened a pull request for improving the performance of legacy main resolve in module resolution: src,lib: reducing C++ calls of esm legacy main resolve node#48325

Author: ```suggestion
* Vinícius opened a pull request for improving the performance of legacy main resolve in module resolution: src,lib: reducing C++ calls of esm legacy main resolve node#48325
```
